src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -144,13 +150,23 @@ private static Collection<IdCredentials> fetchCredentials() {
                     }
                     case "sshUserPrivateKey": {
                         String usernameSecretTag = tags.get("username-is-secret");
+                        String passphraseID = tags.get("passphrase-id");
+                        Secret passphrase = null;
                         boolean usernameSecret = false;
                         if (StringUtils.isNotBlank(usernameSecretTag)) {
                             usernameSecret = Boolean.parseBoolean(usernameSecretTag);
                         }
+                        if (StringUtils.isNotBlank(passphraseID)) {
+                            try {
+                                passphrase = new KeyVaultSecretRetriever(client, keyVaultURL + "secrets/" + passphraseID).get();
+                            } catch (Exception e) {
+                                LOG.log(Level.WARNING, "Could not find passphrase " + passphraseID + " in KeyVault.  Defaulting to null");
```

Comment:
best to not leak code details into user facing error messages, (but as mentioned in https://github.com/jenkinsci/azure-keyvault-plugin/pull/171/files#r1139049840 I don't think we should load a broken secret)
```suggestion
                                LOG.log(Level.WARNING, "Could not find passphrase with ID " + passphraseID + " in KeyVault.  Defaulting to no passphrase");
```

Reply:
Agreed.  I'll update it to just mention that no passphrase with ID could be found, and then skip the iteration entirely.